Bluetooth: Controller: df: Fix lack of IQ reports when IQ sampling enabled and received PDU includes CTE

ppryga-nordic opened the pull request:
The radio_df_cte_ready function returns state of EVENTS_CTEPRESENT.
The function may not be used in create_iq_report function, because
it is called after lll_isr_rx_status_reset. The lll_isr_rx_status-
_reset clears EVENTS_CTEPRESENT value.

The result of this calls chain is lack of IQ sample report no matter
if the PDU has CTEInfo and radio has collected IQ samples.

To fix the problem, state of EVENTS_CTEPRESENT has to be get when
e.g. rssi_ready is get.

Second thing that is change by this PR is related with a EVENTS_CTEPRESENT.
Clear of the event was compiled only for DF connected mode.
It should be compiled in for both connectionless and connected mode.

ppryga-nordic committed:
Bluetooth: Controller: lll: df: Check CTEPRESENT before radio reset

The radio_df_cte_ready function returns state of EVENTS_CTEPRESENT.
The function may not be used in create_iq_report function, because
it is called after lll_isr_rx_status_reset. The lll_isr_rx_status-
_reset clears EVENTS_CTEPRESENT value.

The result of this calls chain is lack of IQ sample report no matter
if the PDU has CTEInfo and radio has collected IQ samples.

To fix the problem, state of EVENTS_CTEPRESENT has to be get when
e.g. rssi_ready is get.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
Bluetooth: Conroller: radio: Clear EVENTS_CTEPRESENT always if DF supp

The EVENTS_CTEPRESENT is available when radio peripheral has
direction finding support. The event is set if received
PDU has CTEInfo and there were collected IQ samples during
CTE reception.

The event should be cleared always when other events are cleared.
That guarantees that the event is not set when use of radio for
following activities.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:
```diff
@@ -52,7 +52,7 @@ static int prepare_cb_common(struct lll_prepare_param *p, uint8_t chan_idx);
 static int is_abort_cb(void *next, void *curr, lll_prepare_cb_t *resume_cb);
 static void abort_cb(struct lll_prepare_param *prepare_param, void *param);
 static int isr_rx(struct lll_sync *lll, uint8_t node_type, uint8_t crc_ok, uint8_t rssi_ready,
-		  enum sync_status status);
+		  enum sync_status status, uint8_t cte_ready);
```
Again my preference is to have the `cte_ready` before `status` or before `rssi_ready`, as it is radio related.

ppryga replied:
I've moved `cte_ready` before `rssi_ready`.

cvinayak reviewed (dismissed):
Minor suggestions, otherwise looks good.

ppryga commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:
```diff
@@ -730,8 +741,14 @@ static void isr_rx_adv_sync_estab(void *param)
 		sync_ok = sync_filtrate_by_cte_type(lll->cte_type, lll->filter_policy);
 		trx_cnt = 1U;
+
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
```
Changed to use `IS_EANBLED`

ppryga-nordic committed:
Bluetooth: Controller: df: make radio_df_vendor_hal.h compile in always

The header file radio_df_vendor_hal.h was conditionally copiled in.
There were a dependency on a CONFIG_BT_CTRL_DR. The dependency is
removed because radio_df_vendor_hal.h and radio_df.h hasn't got any
DF related type dependencies. What more it allows to use IS_ENABLED
for code that uses types and functions defined in radio_df.h.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
I've added one new commit that removes `#if defined(CONFIG_BT_CTLR_DF)` from `subsys/bluetooth/controller/hal/radio_df.h`.
That prevented `subsys/bluetooth/controller/ll_sw/nordic/hal/radio/radio_df.h` from being included. As a result of that compiler was complaining about implicit declaration of `radio_df_cte_ready()`.
